### ISSUE TITLE
Ruby script doesn't work

### ISSUE THREAD

**Issue by shiba89** [woodruffw/ff2mpv#1]
Following brian_ comment on the addon page, the Ruby script exits with errors. Relevant output from Firefox debugger:
```
stderr output from native app ff2mpv: /usr/lib64/ruby/2.2.0/json/common.rb:155:in `parse': 776: unexpected token at '"https://www.youtube.com/watch?v=9LK1jpkDh5Y"' (JSON::ParserError)
stderr output from native app ff2mpv: 	from /usr/lib64/ruby/2.2.0/json/common.rb:155:in `parse'
stderr output from native app ff2mpv: 	from /usr/local/bin/ff2mpv:7:in `<main>'
Error: An unexpected error occurred
```
As already pointed out, it works fine using the Python script from that other project.

**Comment by woodruffw** [comment]
I think this is a bug in Ruby's JSON module. On Ruby 2.4, this works correctly:

```ruby
>> JSON.parse '"foo"' # => "foo"
```

But it fails on 2.0 (and I'm guessing 2.2 as well):

```ruby
>> JSON.parse '"foo"'
JSON::ParserError: 757: unexpected token at '"foo"'
```

I'm going to report this upstream, but I'll work around it for the time being by sending a hash payload instead of just a string literal.

Thanks for reporting!

**Comment by woodruffw** [comment]
Released as 2.1 on AMO: https://addons.mozilla.org/en-US/firefox/addon/ff2mpv/versions/2.1

Please remember to use the latest version of the Ruby script as well!

**Comment by shiba89** [comment]
Sorry to bother you again, but I upgraded everything and now mpv launches but dies immediately if i don't remove `out: :close` from https://github.com/woodruffw/ff2mpv/blob/ceead00e29dcbbc312a2f1fe1d95e492a49ac213/ff2mpv#L9
I've never written a line of code in Ruby, so I have no idea what I just did 😅

**Comment by woodruffw** [comment]
`out: :close` just tells `spawn` to close `stdout` immediately, instead of leaving it open.

I guess it could be a compatibility/configuration issue -- what's your `mpv --version` and `~/.mpv/config` (or `~/.config/mpv/config`)? Also, are you on Linux?

**Comment by shiba89** [comment]
Now I'm taking a random guess, not scientifically proven at all. If I add `--no-terminal` (which according to the man disables std{in,out,err}) as an mpv option in the Ruby script, it all seems to work fine. This leads me to think that MAYBE on my 11yo system with a 5400RPM HDD iostreams get closed before mpv actually starts and it doesn't like it very much?

Anyway I'm on Gentoo, mpv 0.25.0. Even upgrading to 0.27.0 doesn't seem to make any difference. My `.config/mpv/mpv.conf` (not so much relevant I think):
```
sub-scale-with-window=no
sub-use-margins=no
sub-font="Source Sans Pro Semibold"
sub-font-size=58
sub-color="#FFFFFFFF"
sub-border-color="#FF262626"
sub-border-size=3.2
sub-shadow-offset=1
sub-shadow-color="#33000000"
sub-spacing=0.5

deinterlace=yes
```

**Comment by woodruffw** [comment]
> Now I'm taking a random guess, not scientifically proven at all. If I add --no-terminal (which according to the man disables std{in,out,err}) as an mpv option in the Ruby script, it all seems to work fine. This leads me to think that MAYBE on my 11yo system with a 5400RPM HDD iostreams get closed before mpv actually starts and it doesn't like it very much?

Sounds like a good guess to me. Since we're closing all three streams anyways, I'll add that `--no-terminal` flag to the `mpv` invocation. Thanks for taking the time test it out!

I wonder why I haven't experienced that behavior. Maybe `--force-window` or one of my own settings masks it.
